lightning/src/chain/channelmonitor.rs

```diff
@@ -2090,7 +2090,13 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 		self.last_block_hash = block_hash;
 		self.last_block_height = height;
 
-		self.block_confirmed(height, vec![], vec![], vec![], broadcaster, fee_estimator, logger)
+		if height > self.last_block_height {
```

Comment:
FYI, this is wrong since `self.last_block_height` is set to `height` above. No tests for decreasing height yet so unfortunately not caught. Will be fixed in my next push.

Reply:
Actually, perhaps the tests should have caught this. The setup might not be capturing the behavior that would have caused this to fail.

Reply:
I tried running the functional tests by changing the default `ConnectionStyle` (thanks for the suggestion @TheBlueMatt ). I'm able to get `ln::functional_tests::test_htlc_on_chain_timeout` to fail under `ConnectStyle::BestBlockFirstSkippingBlocks` and `ConnectStyle::TransactionsFirstSkippingBlocks` prior to the bug fix and passing after it.

```term
thread 'ln::functional_tests::test_htlc_on_chain_timeout' panicked at 'assertion failed: `(left == right)`
  left: `0`,
 right: `1`', lightning/src/ln/functional_tests.rs:3040:9
```

https://github.com/rust-bitcoin/rust-lightning/blob/c4297bcccfcbb6f12276dcef3c494ae81bb6284a/lightning/src/ln/functional_tests.rs#L3040

Reply:
Can we add those connection types in `test_htlc_on_chain_timeout` to find any regressions?

Reply:
Done.